tests/ui/methods.rs

```diff
@@ -138,6 +141,21 @@ impl<T> V<T> {
     }
 }
 
+struct AsyncNew;
+
+impl AsyncNew {
+    fn new() -> impl Future<Output = Option<Self>> {
```

Comment:
Why not just make this function `async` and use the `async_await` feature in this test?

Everything else LGTM.

Reply:
Because the tests seem to use Rust 2015 (which does not have async fns) and I don't know how to change that.

Reply:
Ah, this again :smile:

You have to add
https://github.com/rust-lang/rust-clippy/blob/d71e9c4f10da31bbdefa4dad64b1f38a456c89ce/tests/ui/issue_4266.rs#L1-L2
at the top of the test case. We should document this.